src/meshapi/views/geography.py

```diff
@@ -365,3 +372,91 @@ def get(self, request: HttpRequest) -> HttpResponse:
             content_type=KML_CONTENT_TYPE_WITH_CHARSET,
             status=status.HTTP_200_OK,
         )
+
+
+@dataclass
+class GeocodeRequest:
+    street_address: str
+    city: str
+    state: str
+    zip: int
+
+
+class GeocodeSerializer(DataclassSerializer):
+    class Meta:
+        dataclass = GeocodeRequest
+
+
+@extend_schema_view(
+    get=extend_schema(
+        tags=["Geographic & KML Data"],
+        summary="Use the NYC geocoding APIs to look up an address, and return the lat/lon/alt "
+        "corresponding to it or 404 if the address cannot be found within NYC",
+        parameters=[GeocodeSerializer],
+        responses={
+            "201": OpenApiResponse(
+                inline_serializer(
+                    "JoinFormSuccessResponse",
+                    fields={
+                        "BIN": serializers.IntegerField(),
+                        "latitude": serializers.FloatField(),
+                        "longitude": serializers.FloatField(),
+                        "altitude": serializers.FloatField(required=False),
+                    },
+                ),
+                description="Request received, an install has been created (along with member and "
+                "building objects if necessary).",
+            ),
+            "400": OpenApiResponse(
+                inline_serializer("ErrorResponseMissingFields", fields={"detail": serializers.DictField()}),
+                description="Invalid request body JSON or missing required fields",
+            ),
+            "404": OpenApiResponse(
+                inline_serializer("ErrorResponseInvalidAddr", fields={"detail": serializers.CharField()}),
+                description="Invalid address, or not found within NYC",
+            ),
+            "500": OpenApiResponse(
+                inline_serializer("ErrorResponseInternalFailure", fields={"detail": serializers.CharField()}),
+                description="Could not geocode address due to internal failure. Try again?",
+            ),
+        },
+    )
+)
+class NYCGeocodeWrapper(APIView):
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request: Request) -> Response:
+        serializer = GeocodeSerializer(data=request.query_params)
+        if not serializer.is_valid():
+            return Response({"detail": serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            raw_addr: GeocodeRequest = serializer.save()
+            nyc_addr_info = geocode_nyc_address(raw_addr.street_address, raw_addr.city, raw_addr.state, raw_addr.zip)
+        except ValueError:
+            return Response(
+                {
+                    "detail": "Non-NYC registrations are not supported at this time. Check back later, "
```

Comment:
Small Nit: Maybe we shouldn't say "check back later." I know we've been doing that for like a year, but it's probably more productive to just say "contact support."

Reply:
I can change this, but really this should be handled in the frontend and nobody should see this message